src/ansys/aedt/core/application/aedt_file_management.py

```diff
@@ -199,26 +209,24 @@ def change_model_orientation(origfile, bottom_dir):
         "-100, 100, -100, 100, -100, 100) ",
     }
 
-    if not os.path.isfile(origfile + ".lock"):  # check if the project is closed
-        # Opening files
-        with open(origfile, "rb") as f, open(newfile, "wb") as n:
-            # Reading file content
-            content = f.read()
+    if lock_file.exists():  # pragma: no cover
+        logging.error(f"change_model_orientation: Project {input_file} is still locked.")
+        return False
 
-            # Searching file content for pattern
-            pattern = re.compile(
-                r"(\$begin 'EditorWindow'\n.+?)(OrientationMatrix\(.+?\))(.+\n\s*\$end 'EditorWindow')", re.UNICODE
-            )
+    try:
+        with input_path.open("rb") as f, newfile.open("wb") as n:
+            content = f.read()
+            search_str = r"(\\$begin 'EditorWindow'\\n.+?)(OrientationMatrix\\(.+?\\))(.+\\n\\s*\\$end 'EditorWindow')"
             # Replacing string
             orientation_str = orientation[bottom_dir]
-            s = pattern.sub(r"\1" + orientation_str + r"\3", content)
-
+            sub_str = r"\1" + orientation_str + r"\3"
+            s = re.sub(search_str.encode("utf-8"), sub_str.encode("utf-8"), content)
             # Writing file content
-            n.write(str(s))
-
-        # Renaming files and deleting temp
-        os.remove(origfile)
-        os.rename(newfile, origfile)
-
-    else:  # Project is locked
-        print("change_model_orientation: Project %s is still locked." % origfile)
+            n.write(s)
+        input_path.unlink()
+        newfile.rename(input_path)
+        return True
+    except Exception as e:  # pragma: no cover
+        if newfile.exists():
+            newfile.unlink()
```

Comment:
In Python 3.8+ the following should be the most pythonic way.
```suggestion
        newfile.unlink(missing_ok=True)
```